src/ServiceControl.Audit/Infrastructure/RavenDB/MigrateKnownEndpoints.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace ServiceControl.Audit.Infrastructure.RavenDB
+{
+    using NServiceBus;
+    using NServiceBus.Features;
+    using Raven.Abstractions.Data;
+    using Raven.Client;
+    using ServiceControl.Audit.Monitoring;
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    class MigrateKnownEndpoints : Feature
+    {
+        public MigrateKnownEndpoints()
+        {
+            EnableByDefault();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            context.RegisterStartupTask(b => new KnownEndpointMigrator(b.Build<IDocumentStore>()));
+        }
+    }
+
+    class KnownEndpointMigrator : FeatureStartupTask
+    {
+        IDocumentStore store;
+
+        public KnownEndpointMigrator(IDocumentStore store)
+        {
+            this.store = store;
+        }
+
+        protected override async Task OnStart(IMessageSession messageSession)
+        {
+            var knownEndpointsIndex = await store.AsyncDatabaseCommands.GetIndexAsync("EndpointsIndex").ConfigureAwait(false);
+            if (knownEndpointsIndex == null)
+            {
+                return;
+            }
+
+            await WaitForNonStaleIndex(knownEndpointsIndex.Name).ConfigureAwait(false);
+
+            int previouslyDone = 0;
+            do {
+                using (var session = store.OpenAsyncSession())
+                {
+                    var endpointsFromIndex = await session.Query<dynamic>(knownEndpointsIndex.Name, true)
+                        .Take(1024)
+                        .Skip(previouslyDone)
```

Comment:
Shouldn't this be `Skip` then `Take`?

Reply:
The order doesn't matter until the expression is evaluated, but I've change it around for readability

Reply:
That's Raven weirdness then as order should matter. `Enumerable.Range(1, 6).Take(2).Skip(2)` returns an empty collection, `Enumerable.Range(1,6).Skip(2).Take(2)` returns `[3, 4]`